From d1381835c3017c44b987245f503e7893882a8d8a Mon Sep 17 00:00:00 2001
From: Brendan Dolan-Gavitt <brendandg@gatech.edu>
Date: Sun, 11 Oct 2015 16:09:54 -0400
Subject: [PATCH 02/13] Allow printing unsigned long long in cprintf

---
 console.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/console.c b/console.c
index 35f221d..1012fb7 100644
--- a/console.c
+++ b/console.c
@@ -47,6 +47,17 @@ printint(int xx, int base, int sign)
   while(--i >= 0)
     consputc(buf[i]);
 }
+
+static void
+printlong(unsigned long long xx, int base, int sgn)
+{
+    // Force hexadecimal
+    uint upper, lower;
+    upper = xx >> 32;
+    lower = xx & 0xffffffff;
+    if(upper) printint(upper, 16, 0);
+    printint(lower, 16, 0);
+}
 //PAGEBREAK: 50
 
 // Print to the console. only understands %d, %x, %p, %s.
@@ -77,6 +88,12 @@ cprintf(char *fmt, ...)
     case 'd':
       printint(*argp++, 10, 1);
       break;
+    case 'l':
+        printlong(*(unsigned long long *)argp, 10, 0);
+        // long longs take up 2 argument slots
+        argp++;
+        argp++;
+        break;
     case 'x':
     case 'p':
       printint(*argp++, 16, 0);
-- 
1.9.1

